frontend/packages/dev-console/src/components/edit-application/EditApplication.tsx

```diff
@@ -1,113 +1,104 @@
-      : null;
-
-  const currentImage = _.split(
-    _.get(appResources, 'buildConfig.data.spec.strategy.sourceStrategy.from.name', ''),
-    ':',
-  );
-
-  const appGroupName = _.get(editAppResource, 'metadata.labels["app.kubernetes.io/part-of"]');
-
-  const initialValues = {
-    formType: 'edit',
-    name: appName,
-    application: {
-      name: appGroupName,
-      selectedKey: appGroupName,
-    },
-    project: {
-      name: namespace,
-    },
-    git: EditApplicationUtils.getGitData(_.get(appResources, 'buildConfig.data')),
-    docker: {
-      dockerfilePath: _.get(
-        appResources,
-        'buildConfig.data.spec.strategy.dockerStrategy.dockerfilePath',
-        'Dockerfile',
-      ),
-      containerPort: parseInt(
-        _.split(_.get(appResources, 'route.data.spec.port.targetPort'), '-')[0],
-        10,
-      ),
-    },
-    image: {
-      selected: currentImage[0] || '',
-      recommended: '',
-      tag: currentImage[1] || '',
-      tagObj: {},
-      ports: [],
-      isRecommending: false,
-      couldNotRecommend: false,
-    },
-    route: EditApplicationUtils.getRouteData(_.get(appResources, 'route.data'), editAppResource),
-    resources: EditApplicationUtils.getResourcesType(editAppResource),
-    serverless: EditApplicationUtils.getServerlessData(editAppResource),
-    pipeline: {
-      enabled: false,
-    },
-    build: EditApplicationUtils.getBuildData(_.get(appResources, 'buildConfig.data')),
-    deployment: EditApplicationUtils.getDeploymentData(editAppResource),
-    labels: EditApplicationUtils.getUserLabels(editAppResource),
-    limits: EditApplicationUtils.getLimitsData(editAppResource),
+      ? appResources.imageStreams.data
+      : [];
+  const builderImages: NormalizedBuilderImages = !_.isEmpty(imageStreamsData)
+    ? normalizeBuilderImages(imageStreamsData)
+    : null;
+  const initialValues = getInitialValues(appResources, appName, namespace);
+  const pageHeading = getPageHeading(_.get(initialValues, 'build.strategy', ''));
+  const handleRedirect = (project: string) => {
+    const perspectiveData = plugins.registry
+      .getPerspectives()
+      .find((item) => item.properties.id === perspective);
+    const redirectURL = perspectiveData.properties.getImportRedirectURL(project);
+    history.push(redirectURL);
   };
-
   const handleSubmit = (values, actions) => {
-    const imageStream =
-      values.image.selected && builderImages ? builderImages[values.image.selected].obj : null;
-
-    createOrUpdateResources(
-      values,
-      imageStream,
-      false,
-      false,
-      'update',
-      appResources,
-      editAppResource,
-    )
-      .then(() => {
-        actions.setSubmitting(false);
-        actions.setStatus({ error: '' });
-        onSubmit();
-      })
-      .catch((err) => {
-        actions.setSubmitting(false);
-        actions.setStatus({ error: err.message });
-      });
+    if (!_.isEmpty(values.build.strategy)) {
```

Comment:
This would be better since this is just a string check.
```suggestion
    if (values.build.strategy) {
```